src/elements/Asset.php

```diff
@@ -1053,6 +1053,19 @@ private static function isFolderIndex(): bool
      */
     private ?int $_oldVolumeId = null;
 
+    public function __construct($config = [])
+    {
+        // make sure asset folderPath accounts for volume's fsSubpath
+        if (isset($config['volumeId'])) {
+            $volume = Craft::$app->getVolumes()->getVolumeById($config['volumeId']);
+            $fsSubpath = $volume?->getFsSubpath();
+            if (!empty($fsSubpath) && !str_starts_with($config['folderPath'], $fsSubpath)) {
+                $config['folderPath'] = $fsSubpath . $config['folderPath'];
+            }
+        }
```

Comment:
Similar to my [previous comment](https://github.com/craftcms/cms/pull/12173/files/dc6fdac4c3285da3e43bbf6c48100c65bbe6c5a2#diff-eb148c3d8c11461762a09bac958055437d85ede94dfbca9380d33a0de7483757) – asset folder paths should be relative to the root of the _volume_ (like folder paths), **not** to the root of the filesystem.

Reply:
Basically, nothing about working with assets should reveal anything about the fact that a filesystem subpath was specified on the volume settings. All of the filesystem subpath logic should be completely contained to the volume itself.